### PR TITLE
Add Rendmaw Creaking Nest and supporting changes

### PR DESCRIPTION
**Pull request by TwentyToedToad**
Added a Black 2/2 Bird Creature with Flying token
Added a property to check for number of types a card has

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/rendmaw_creaking_nest.txt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Name:Rendmaw, Creaking Nest
+ManaCost:3 B G
+Types:Legendary Artifact Creature Scarecrow
+PT:5/5
+K:Menace
+K:Reach
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ DBTokens | TriggerDescription$ When CARDNAME enters each player creates a tapped 2/2 black Bird creature token with flying. The tokens are goaded for the rest of the game.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/rendmaw_creaking_nest.txt:7]: when a trigger is caused by 2+ different events we still use the printed description
but we add Secondary$ param after the first one so it doesn't show up ingame

File: forge-game/src/main/java/forge/game/card/CardProperty.java [hunk 2]
```diff
@@ -1529,6 +1526,9 @@ else if (prop.isEmpty() && dmgSource.equalsWithGameTimestamp(source)) {
             } else if (property.startsWith("numColors")) {
                 rhs = property.substring(11);
                 y = card.getColor().countColors();
+            } else if (property.startsWith("numTypes")) {
+                rhs = property.substring(10);
+                y = ((Collection<?>)card.getType().getCoreTypes()).size();
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/card/CardProperty.java:1531]: this is bad for type safety
use `Iterables.size`

Comment (TwentyToedToad) [reply forge-game/src/main/java/forge/game/card/CardProperty.java]: Big mistake, thanks for the catch!

File: forge-gui/res/cardsfolder/upcoming/rendmaw_creaking_nest.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/rendmaw_creaking_nest.txt]: both need the same TriggerDescription
(relevant if e.g. you're asked to order simultaneous stack entry)

**Review by tool4ever — APPROVED** [review]

**Pull request merged**